src/gt4py/backend/gtc_backend/defir_to_gtir.py

```diff
@@ -1,38 +1,55 @@
 from types import MappingProxyType
 from typing import ClassVar, Dict, List, Mapping
-from gt4py.backend.gtc_backend import gtir, common
+
+from gt4py.backend.gtc_backend import common, gtir
 from gt4py.ir import IRNodeVisitor
-from gt4py.ir.nodes import Assign, AxisBound, AxisInterval, BinOpExpr, BinaryOperator, BlockStmt, ComputationBlock, FieldRef, ScalarLiteral, StencilDefinition, LevelMarker, FieldDecl, IterationOrder
+from gt4py.ir.nodes import (
+    Assign,
+    AxisBound,
+    AxisInterval,
+    BinaryOperator,
+    BinOpExpr,
+    BlockStmt,
+    ComputationBlock,
+    FieldDecl,
+    FieldRef,
+    IterationOrder,
+    LevelMarker,
+    ScalarLiteral,
+    StencilDefinition,
+)
 
 
 def transform_offset(offset: Dict[str, int]) -> gtir.CartesianOffset:
-    i = offset['I'] if 'I' in offset else 0
-    j = offset['J'] if 'J' in offset else 0
-    k = offset['K'] if 'K' in offset else 0
+    i = offset["I"] if "I" in offset else 0
+    j = offset["J"] if "J" in offset else 0
+    k = offset["K"] if "K" in offset else 0
     return gtir.CartesianOffset(i=i, j=j, k=k)
 
 
+# TODO(Rico Häuselmann): write unit tests
+
+
 class DefIRToGTIR(IRNodeVisitor):
     @classmethod
     def apply(cls, root, **kwargs):
         return cls().visit(root)
 
-    GT4PY_ITERATIONORDER_TO_GTIR_LOOPORDER: ClassVar[Mapping[IterationOrder, common.LoopOrder]] = MappingProxyType(
+    GT4PY_ITERATIONORDER_TO_GTIR_LOOPORDER: ClassVar[
+        Mapping[IterationOrder, int]
+    ] = MappingProxyType(
         {
             IterationOrder.BACKWARD: common.LoopOrder.BACKWARD,
             IterationOrder.PARALLEL: common.LoopOrder.PARALLEL,
-            IterationOrder.FORWARD: common.LoopOrder.FORWARD
+            IterationOrder.FORWARD: common.LoopOrder.FORWARD,
         }
     )
 
-    GT4PY_LEVELMARKER_TO_GTIR_LEVELMARKER: ClassVar[Mapping[LevelMarker, common.LevelMarker]] = MappingProxyType(
-        {
-            LevelMarker.START: common.LevelMarker.START,
-            LevelMarker.END: common.LevelMarker.END
-        }
+    GT4PY_LEVELMARKER_TO_GTIR_LEVELMARKER: ClassVar[Mapping[LevelMarker, str]] = MappingProxyType(
+        {LevelMarker.START: common.LevelMarker.START, LevelMarker.END: common.LevelMarker.END}
     )
 
-    GT4PY_OP_TO_GTIR_OP: ClassVar[Mapping[BinaryOperator, common.BinaryOperator]] = MappingProxyType(
+    GT4PY_OP_TO_GTIR_OP: ClassVar[Mapping[BinaryOperator, str]] = MappingProxyType(
```

Comment:
The most pythonic way to implement a switch is a dict: `{<switch-value1>: <outcome1>, ...}`

Reply:
In case it helps, this is an example of how this is currently implemented (using a dict, as Rico mentions) in one of the unstructured code generators: https://github.com/GridTools/gtc/blob/15e835c6df1a81ef591814acce3e13ba7de915d9/src/gtc/unstructured/naive_codegen.py#L54